cpp/src/strings/replace/multi.cu

```diff
@@ -0,0 +1,488 @@
+
+    return make_strings_column(input.size(),
+                               std::move(children.first),
+                               std::move(children.second),
+                               input.null_count(),
+                               cudf::detail::copy_bitmask(input.parent(), stream, mr));
+  }
+
+  auto const strings_count = input.size();
+  auto const chars_bytes =
+    cudf::detail::get_value<size_type>(input.offsets(), input.offset() + strings_count, stream) -
+    cudf::detail::get_value<size_type>(input.offsets(), input.offset(), stream);
+
+  auto d_targets =
+    create_string_vector_from_column(targets, stream, rmm::mr::get_current_device_resource());
+  auto d_replacements =
+    create_string_vector_from_column(repls, stream, rmm::mr::get_current_device_resource());
+
+  replace_multi_parallel_fn fn{*d_strings, d_targets, d_replacements};
+
+  // count the number of targets in the entire column
+  auto const target_count = thrust::count_if(rmm::exec_policy(stream),
+                                             thrust::make_counting_iterator<size_type>(0),
+                                             thrust::make_counting_iterator<size_type>(chars_bytes),
+                                             [fn, chars_bytes] __device__(size_type idx) {
+                                               return fn.has_target(idx, chars_bytes).has_value();
+                                             });
+  // Create a vector of every target position in the chars column.
+  // These may include overlapping targets which will be resolved later.
+  auto targets_positions = rmm::device_uvector<target_pair>(target_count, stream);
+  auto d_positions       = targets_positions.data();
+
+  auto const copy_itr =
+    cudf::detail::make_counting_transform_iterator(0, pair_generator{fn, chars_bytes});
+  auto const copy_end = thrust::copy_if(
+    rmm::exec_policy(stream), copy_itr, copy_itr + chars_bytes, d_positions, copy_if_fn{});
+
+  // create a vector of offsets to each string's set of target positions
+  auto const targets_offsets = [&] {
+    auto string_indices = rmm::device_uvector<size_type>(target_count, stream);
+
+    auto const pos_itr = cudf::detail::make_counting_transform_iterator(
+      0, [d_positions] __device__(auto idx) -> size_type { return d_positions[idx].first; });
+    auto pos_count = std::distance(d_positions, copy_end);
+
+    thrust::upper_bound(rmm::exec_policy(stream),
+                        input.offsets_begin(),
+                        input.offsets_end(),
+                        pos_itr,
+                        pos_itr + pos_count,
+                        string_indices.begin());
+
+    // compute offsets per string
+    auto targets_offsets   = rmm::device_uvector<size_type>(strings_count + 1, stream);
+    auto d_targets_offsets = targets_offsets.data();
+
+    // memset to zero-out the target counts for any null-entries or strings with no targets
+    CUDF_CUDA_TRY(cudaMemsetAsync(
```

Comment:
Developer guide shows use of `thrust::uninitialized_fill` to initialize a `device_uvector`. Should that be used here?